core/app/models/spree/user_address.rb

```diff
@@ -4,12 +4,18 @@ class UserAddress < Spree::Base
     belongs_to :address, class_name: "Spree::Address"
 
     validates_uniqueness_of :address_id, scope: :user_id
-    validates_uniqueness_of :user_id, conditions: -> { where(default: true) }, message: :default_address_exists, if: :default?
+    validates_uniqueness_of :user_id, conditions: -> { where(default: true, archived: false) }, message: :default_address_exists, if: :default?
 
     scope :with_address_values, ->(address_attributes) do
       joins(:address).merge(
         Spree::Address.with_values(address_attributes)
       )
     end
+
+    scope :all_historical, ->() do
```

Comment:
Don't left empty parentes in lambdas.
This scope is so small, that should be in one line.